pkg/kubectl/cmd/util/clientcache.go

```diff
@@ -58,6 +59,34 @@ type ClientCache struct {
 	// argument evaluation
 	discoveryClientFactory DiscoveryClientFactory
 	discoveryClient        discovery.DiscoveryInterface
+
+	kubernetesClientLoader KubernetesClientCache
+}
+
+// kubernetesClientLoader provides a new kubernetes.Clientset
+type KubernetesClientCache struct {
+	// once makes sure the client is only initialized once
+	once sync.Once
+	// client is the cached client value
+	client *kubernetes.Clientset
+	// err is the cached error value
```

Comment:
What is the meaning of `err != nil`? If `err != nil` can any future calls succeed?